Core/Src/analyzer.c

```diff
@@ -613,47 +604,13 @@ uint8_t analyzer_calc_fan_pwm()
 	       (2 * 5);
 }
 
-void analyzer_push(acc_data_t *data)
-{
-	if (prevbmsdata != NULL)
-		free(bmsdata);
-
-	prevbmsdata = bmsdata;
-	bmsdata = data;
-
-	disable_therms();
-
-	//high_curr_therm_check(); /* = prev if curr > 50 */
-	// diff_curr_therm_check();     /* = prev if curr - prevcurr > 10 */
-	// variance_therm_check();      /* = prev if val > 5 deg difference */
-	// standard_dev_therm_check();  /* = prev if std dev > 3 */
-	// averaging_therm_check();     /* matt shitty incrementing */
-
-	calc_cell_temps();
-	calc_pack_temps();
-	calc_pack_voltage_stats();
-	calc_open_cell_voltage();
```

Comment:
i still think one thread that does all of this is best. Time sync becomes really hard to follow. We poll data rarely enough (every 500 ms) that we arent at risk of starving that thread if we let this be one thread calcualting all of it. By mkaing it differernt threads, you run the risk of having some calculations done on data pulled at one time, and then some calcs done after the next reading. We dont want that; keep this simple and do all simple calcs that involve very little processing together

Reply:
Fixed